packages/utils/src/path.ts

```diff
@@ -51,7 +51,7 @@ function normalizeArray(parts: string[], allowAboveRoot?: boolean): string[] {
 
 // Split a filename into [root, dir, basename, ext], unix version
 // 'root' is just a slash, or nothing.
-const splitPathRe = /^(\/?|)([\s\S]*?)((?:\.{1,2}|[^/]+?|)(\.[^./]*|))(?:[/]*)$/;
+const splitPathRe = /^(\S+:\\|\/?)([\s\S]*?)((?:\.{1,2}|[^/\\]+?|)(\.[^./\\]*|))(?:[/\\]*)$/;
 /** JSDoc */
 function splitPath(filename: string): string[] {
   const parts = splitPathRe.exec(filename);
```

Comment:
@timfish it complains about the Regex here, not sure if this _actually- got worse or we just ignored this warning before...?

Reply:
Yep I noticed that. I'm not convinced it's any worse than before but I can't see the list of reviously ignored warnings

Reply:
Yeah not really much detail in the history, it was also ignored the last time 😅 so I think we can dismiss this, unless cc @lforst or @Lms24 have some other opinion?

Reply:
Polynomial regexes always confuse the hell out of me. In this case I wouldn't know how to fix it. I like this take right here: https://codeql.github.com/codeql-query-help/javascript/js-polynomial-redos/#id2

If we limit the length of `filename` to any constant length we drastically limit to expose anybody to a ddos by collapsing the polynomial runtime to a constant. Imo something like 500 char length is very unlikely gonna be hit by ppl and 500^2 is 250k which should be not too expensive to compute in case of an attack (?).

Reply:
I don't like dismissing it tbh. It poses a security risk. I don't know the historical context though.

Reply:
limiting it to a high max. length sounds like a reasonable & easy enough approach to me!

Reply:
> limit the length of filename to any constant length 

Yep, I already added this for the stack parsers at 1k:
https://github.com/getsentry/sentry-javascript/blob/e0e246d42670a7cc331f87f22a33a9a3ba2756ad/packages/utils/src/stacktrace.ts#L27-L33

So 500 characters for `filename` should be ok?

Reply:
> So 500 characters for filename should be ok?

Yep! Just wondering if we should add some kind of indicator that the filename has been truncated, or that we bailed, just so debugging gets a little bit easier if people have troubles with their source maps.

Reply:
Yeah probably a good idea, something like?

```ts
function splitPath(filename: string): string[] {
  if(filename.length > 500) {
    filename = '<truncated>' + filename.slice(-500);
  }

  // etc
}
```

Reply:
We have the `errors` field on events: https://develop.sentry.dev/sdk/event-payloads/ but this might be a lot of work...

Let's just go for it without any logs on the event for now. Maybe we can add debug logging though.

Reply:
> if I set the filename character limit to 1k rather than 500, we'll be even less likely to hit it since the stack parser will skip and not even parse lines that long.

Seems good!